src/adapter.js

```diff
@@ -39,20 +37,18 @@
         if(!karma.config.jspm.useBundles){
             System.bundles = [];
         }
-        
-        // Load everything specified in loadFiles
-        for (var i = 0; i < karma.config.jspm.expandedFiles.length; i++) {
-            var modulePath = karma.config.jspm.expandedFiles[i];
-            var promise = System['import'](extractModuleName(modulePath))
-                ['catch'](function(e) {
-                    throw e;
-                });
-            promises.push(promise);
-        }
 
-        // Promise comes from the systemjs polyfills
-        Promise.all(promises).then(function() {
+        // Load everything specified in loadFiles in the specified order
+        var promiseChain = karma.config.jspm.expandedFiles.reduce(function (prev, next) {
+            return prev.then(function () {
+                return System.import(extractModuleName(next));
```

Comment:
this needs to use the `System['import']` syntax from above for IE8 compatibility